core/consensus/authority/impl/authority_manager_impl.cpp

```diff
@@ -61,29 +73,38 @@ namespace kagome::authority {
 
   void AuthorityManagerImpl::stop() {
     if (!root_) return;
+    std::ignore = save();
+  }
+
+  outcome::result<void> AuthorityManagerImpl::save() {
+    BOOST_ASSERT(root_ != nullptr);
+
     auto data_res = scale::encode(root_);
     if (!data_res.has_value()) {
       log_->critical("Can't encode state to store");
-      return;
+      return AuthorityManagerError::CAN_NOT_SAVE_STATE;
     }
+
     auto save_res =
         storage_->put(SCHEDULER_TREE, common::Buffer(data_res.value()));
     if (!save_res.has_value()) {
       log_->critical("Can't store current state");
```

Comment:
Maybe also add here the error message from save_res? In my experience error messages which state that something couldn't be done but are silent about why are rather frustrating.

Reply:
Done